annoyed-alligators/socl_media/apps/terminal/commands.py

```diff
@@ -0,0 +1,339 @@
+            try:
+                receiver_obj = User.objects.get(username=receiver_username)
+                if len(params.split()) > 1:
+                    message_body = params[params.find(" ") + 1:]
+                    Message.objects.create(body=message_body,
+                                           sender=request.user,
+                                           recipient=receiver_obj)
+                    message = f"Message sent to {receiver_username}"
+                else:
+                    message = "No message provided. See: message --help"
+
+            except User.DoesNotExist:
+                message = f"User '{receiver_username}' does not exist. Enter a valid username."
+
+    return {'response': add_linebreaks(message)}
+
+
+message = Command(name='message', help_text="""<br>Send a personal message to any user.<br>
+Usage: message [username] [message body]""",
+                  action=message_func)
+
+
+def message_box_func(request, params: str, **kwargs):
+    if not request.user.is_authenticated:
+        message = "Log in to view your Message Box"
+        return {'response': add_linebreaks(message)}
+    else:
+        redirect = reverse('message-box')
+        message = "Here are your received messages"
+        messages.success(request, message)
+        return {'response': add_linebreaks(message), 'redirect': redirect}
+
+
+message_box = Command(name='message-box', help_text="""<br>View your received messages.<br>
+Make sure you are logged in to see your messages.<br>
+Usage: message-box<br>""",
+                      action=message_box_func)
+
+
+def profile_func(request, params: str, **kwargs):
+    if not params:
+        # Show the logged in user's profile if no other username provided
+        message = "Your Profile"
+        messages.success(request, message)
+        redirect = reverse('profile', args=[request.user.username])
+
+    else:
+        username = params.split()[0]
+
+        if username == request.user.username:
+            message = "Your Profile"
+            messages.success(request, message)
+            redirect = reverse('profile', args=[username])
+        else:
+            try:
+                user_obj = User.objects.get(username=username)
+                redirect = reverse('profile', args=[user_obj.username])
+                message = f"{username}'s Profile"
+                messages.success(request, message)
+            except User.DoesNotExist:
+                message = f"User '{username}' does not exist. Enter a valid username."
+                return {'response': add_linebreaks(message)}
+    return {'response': add_linebreaks(message), 'redirect': redirect}
+
+
+profile = Command(name='profile', help_text="""<br>View any user's profile.<br>
+You'll see your own profile if no username is specified.<br>
+Usage: profile [username]""",
+                  action=profile_func)
+
+
+def news_func(request, topic: str, start_date: str = None, end_date: str = None, **kwargs):
```

Comment:
Looks like there are a lot of things going on here. I find it a bit hard to understand.
but that said, nice use of list-comprehension.